### PR TITLE
Removes IAA & IAA pinpointer code improvement

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

This is an alternative to https://github.com/tgstation/tgstation/pull/63897
Since the gamemode isn't wanted here, even just to maintain (and especially now that progression Traitors are being added which will break IAA even more), I'm gonna give my alternate idea to just remove IAA entirely.
This is also to help downstream, as any plan on just 'adding it downstream' is kinda screwed over by it TECHNICALLY still being upstream, and its code conflicting with my own, even if it is just broken and unmaintained.

## Why It's Good For The Game

Removes unwanted, unused and unmaintained gamemode, while still keeping the code improvement for the stuff still used in-game (agent pinpointer is also used by changelings and clarke mechs), allowing downstreams to do any work they would want with the gamemode.

## Changelog

:cl:
code: Internal Affairs gamemode (not the traitor flavor) has been removed.
/:cl:

### CODE REVIEW

**Review by tralezab — APPROVED** [review]
i like it, and well written

**Comment by JohnFulpWillard** [comment]
IAA not even being touched at all by progression traitors really shows how no one cares about them
at least i dont have to solve conflicts :trollface:

File: code/__DEFINES/role_preferences.dm [hunk 1]
```diff
@@ -53,6 +53,7 @@
 #define ROLE_REVENANT "Revenant"
 #define ROLE_BRAINWASHED "Brainwashed Victim"
 #define ROLE_HYPNOTIZED "Hypnotized Victim"
+#define ROLE_INTERNAL_AFFAIRS "Internal Affairs Agent" //Role removed, left here for safety.
```
Comment (Mothblocks) [inline code/__DEFINES/role_preferences.dm:56]: Just remove it

Comment (JohnFulpWillard) [reply code/__DEFINES/role_preferences.dm:56]: :(

**Review by tralezab — APPROVED** [review]

**Pull request merged**